### DISCUSSION TITLE
Firestore not returning maps correctly in android

### DISCUSSION THREAD

**Discussion by Tsquared86** [capawesome-team/capacitor-firebase#496] [Q&A]
Hi I am using the capacitor firestore plugin and it works perfectly on iOS and the web, but on android it returns map objects with an "=" instead of a ":". when this happens the front end reads it as a string and can not iterate it on the frontend.  How can I get past this?

This is what the console logs for the variable format: 
'[{amount=5, name=Eggs, icon=}, {amount=2 slices, icon=, name=Cheese}, {amount=To Taste, icon=, name=Salt}, {amount=To Taste, name=Pepper , icon=}]'

This is the error:
Error: Cannot find a differ supporting object '[{amount=5, name=Eggs, icon=}, {amount=2 slices, icon=, name=Cheese}, {amount=To Taste, icon=, name=Salt}, {amount=To Taste, name=Pepper , icon=}]' of type 'string'. NgFor only supports binding to Iterables such as Arrays.

**Comment by robingenz** [comment]
Thank you for your request. This seems like a bug. Please create a bug report: https://github.com/capawesome-team/capacitor-firebase/issues/new

**Reply by Tsquared86** [reply]
Hey I wanted to but I am unable to provide reproducible code in the form of a GitHub link to fill out the form. Do you have a work around?

**Reply by robingenz** [reply]
> Hey I wanted to but I am unable to provide reproducible code in the form of a GitHub link to fill out the form. Do you have a work around?

This is okay for now. Just insert a "-" as link. Please please include the code you use to store and fetch the document.
